Test/CaptureHandleReaderDeviceTest.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using NUnit.Framework;
+using SharpPcap;
+using SharpPcap.LibPcap;
+using PacketDotNet;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Test
+{
+    [TestFixture]
+    public class CaptureHandleReaderDeviceTest
+    {
+        private static int capturedPackets;
+
+        private SafeHandle GetTestFileHandle(string filename)
+        {
+            filename = TestHelper.GetFile(filename);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // On Windows, we need an actual OS handle.
+                // The FileStream is not disposed, which is alright because we take care of its handle.
+                // On .NET 6+, this can be replaced with File.OpenHandle().
```

Comment:
I could update the project to .NET 6 if that helps simplify the logic, .NET 5 is out of support anyway

Reply:
Yeah, going to .NET 6 can't hurt. It'd allow us to revert 65aa13536cf6e62fe9c39c4df7b69464799e12ca.

Reply:
@heck-gd master branch updated to .net 6

Reply:
Thanks, I rebased the branch to master and switched the code back to the simple variant. Unfortunately this breaks the tests on Windows, due to the conditional `net48` target framework.